src/components/user/createAppointment/Form/FormPatient.js

```diff
@@ -1,48 +1,41 @@
 import { useEffect, useState } from 'react';
 import { CustomSelect } from './Select.styled';
 import { StyledInput, StyledInputContainer, StyledLabel } from './FormPatient.styled';
+import { getAllSpecializations, getDoctorsBySpecializations } from '../../../../store/user/patientOperations';
 
 const FormPatient = ({ onChangeFormData }) => {
-  const [occupationData, setOccupationData] = useState();
-  const [doctorName, setDoctorName] = useState();
+  const [allOccupations, setAllOccupations] = useState([]);
+  const [namesBySpecialization, setNamesBySpecialization] = useState([]);
+  const [doctorId, setDoctorId] = useState();
   const [reason, setReason] = useState();
-  const [note, setNote] = useState();
+  const [note, setNote] = useState('');
 
   useEffect(() => {
+    getAllSpecializations().then((response) => setAllOccupations(response.data.map((item) => (
+      {
+        value: item.id,
+        label: item.specialization_name,
+      }
+    ))));
     const obj = {
-      occupation: occupationData,
-      doctorName,
+      doctorId,
       reason,
       note,
     };
     onChangeFormData(obj);
-  }, [occupationData, doctorName, reason, note]);
-
-  const occupation = [
-    { value: 'cardiologist', label: 'Cardiologist' },
-    { value: 'dermatologist', label: 'Dermatologist' },
-    { value: 'doctor', label: 'Doctor' },
-    { value: 'family Doctor', label: 'Family Doctor' },
-    { value: 'gastroenterologist', label: 'Gastroenterologist' },
-    { value: 'gynecologist', label: 'Gynecologist' },
-    { value: 'surgeon', label: 'Surgeon' },
-    { value: 'therapist', label: 'Therapist' },
-  ];
-
-  const names = [
-    { value: 'john milton', label: 'John Milton' },
-    { value: 'john charles', label: 'John Charles' },
-    { value: 'jonathan meyers', label: 'Jonathan Meyers' },
-    { value: 'andrew story', label: 'Andrew Story' },
-    { value: 'vitalii doc', label: 'Vitalii Doc' },
-  ];
+  }, [doctorId, reason, note]);
 
   const onChangeOccupation = (value) => {
-    setOccupationData(value.value);
+    getDoctorsBySpecializations(value.value).then((response) => {
+      setNamesBySpecialization(response.data.map((item) => ({
+        value: item.id,
+        label: `${item.first_name} ${item.last_name}`,
+      })));
+    });
```

Comment:
Create hook that will resolve this logic out of component.

Reply:
Resolved